edx_solutions_api_integration/imports/views.py

```diff
@@ -0,0 +1,157 @@
+"""
+Views for importing data into the LMS that requires a lot of LMS functionality to process.
+"""
+
+import logging
+
+from django.conf import settings
+from django.contrib.auth.models import Group, User
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.db import IntegrityError
+from django.db.models import Q
+from django.utils.translation import ugettext as _
+from rest_framework.response import Response
+
+from edx_solutions_api_integration.courseware_access import get_course
+from edx_solutions_api_integration.models import CourseGroupRelationship
+from edx_solutions_api_integration.users.views import _manage_role
+from edx_solutions_organizations.models import Organization
+from rest_framework.decorators import list_route
+
+from notification_prefs.views import enable_notifications
+from openedx.core.djangoapps.course_groups.cohorts import add_cohort, get_cohort_by_name
+from openedx.core.djangoapps.course_groups.models import CohortMembership, CourseCohort, CourseUserGroup
+from student.models import CourseEnrollment, PasswordHistory, UserProfile
+
+from edx_solutions_api_integration.permissions import SecureViewSet
+
+AUDIT_LOG = logging.getLogger("audit")
+
+
+class ImportParticipantsViewSet(SecureViewSet):
+
+    @list_route(methods=['post'])
+    def new(self, request):
+        """Import new participants into the LMS, a company and course."""
+        errors, response = [], {}
+        internal = request.data.get('internal', False)
+        statuses = request.data.get('statuses', [])
+        roles = request.data.get('roles', {})
+        ignore_roles = request.data.get('ignore_roles', [])
+        permissions = request.data.get('permissions', {})
+        company_id = request.data.get('company_id', '')
+        course_id = request.data.get('course_id', '')
+        status = request.data.get('status', '').lower()
+        user = request.data.get('user', {})
+        username, email = user.get('username', ''), user.get('email', '')
+
+        # Check for empty fields.
+        for key, value in user.items():
+            if not isinstance(value, bool) and value.strip() == '':
+                if key == 'email':
+                    email = _("No email")
+                if key != 'username':
+                    self._add_error(errors, _("Empty field: {}").format(key), _("Processing Participant"), email)
```

Comment:
Good point. I think because the "No email" text is only useful inside `_add_error`, I'll just remove the special check for email above and let the function itself do the `email or _("No email")` part.